### DISCUSSION TITLE
"Incompatible JVM" message after 23.0.3 install

### DISCUSSION THREAD

**Discussion by JimmyTheExploder** [dbeaver/dbeaver#20163] [Q&A] [answered]
### Description

Getting reports from our users that after they install version 23.0.3 (Community Edition) and launch it, then get a dialog titled "Incompatible JVM". Text in the dialog states "Version 11.0.12 of the JVM is not suitable for this product. Version: 17 or greater is required. 
DBeaver is getting installed via the Company Portal or Software Center on Windows 11/Windows 10 devices. The application is managed by a 3rd party vendor. The installer being used is the x86_64-setup.exe. We've had DBeaver in our environment for at least a couple years and I'm unaware of a similar issue being noted previously. 

I did see a similar issue noted here for version 22.3: https://github.com/dbeaver/dbeaver/issues/18991. 

So far, the fix that has worked for us is to have users reinstall the application. They're then able to launch it without getting the dialog noted above.

### DBeaver Version

Community Edition 23.0.3 

### Operating System

Windows 10 and 11

### Database and driver

DB2 (db2jcc4-4.23.42)
Hive (1.9-2.6.5)

### Steps to reproduce

Install DBeaver from Software Center (WIndows 10) or Company Portal (Windows 11). A previous version of the application (generally 22.2) existed on all devices where users have reported having the issue.
Launch DBeaver.
"Incompatible JVM" dialog appears.

**Comment by hrimthrusa** [comment, marked as answer]
In DBeaver root directory you will find a `dbeaver.ini` file. Add a `-vm` parameter with a correct path to the bin directory of your recent JDK with java version 17 or higher just before the `-vmargs` parameter as shown here:

![1](https://user-images.githubusercontent.com/97725836/236395388-5efc4f64-1b1a-491e-91c2-e75a120a1f76.png)

DBeaver comes with JDK 11.0.18 from the Microsoft Store and you may ask your 3rd party vendor to pack the very same JDK in their repacks.

**Comment by bcag2** [comment]
Similar issue under Ubuntu 22.04 with openjdk 11.0.19 2023-04-18

**Reply by hrimthrusa** [reply]
Did the alteration of _dbeaver.ini_ file help in your case?

**Reply by bcag2** [reply]
No, I tried first to add  
```
-vm
/usr/bin/java
```
always require JAVA 17 instead of 11  

and if I change 
`-Dosgi.requiredJavaVersion=11`  
It returns error in : ~/.eclipse/org.jkiss.dbeaver.product_23.1.2_302967072_linux_gtk_x86_64/configuration/1688989768683.log  
that begin with :  
```
!SESSION 2023-07-10 13:49:28.274 -----------------------------------------------
eclipse.buildId=unknown
java.version=11.0.19
java.vendor=Ubuntu
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=fr_FR
Command-line arguments:  -os linux -ws gtk -arch x86_64

This is a continuation of log file /home/myusername/.eclipse/org.jkiss.dbeaver.product_23.1.2_302967072_linux_gtk_x86_64/configuration/1688989768683.bak_0.log
Created Time: 2023-07-10 13:49:29.791

!ENTRY org.jkiss.dbeaver.ui.statistics 4 0 2023-07-10 13:49:29.791
!MESSAGE FrameworkEvent ERROR
!STACK 0
org.osgi.framework.BundleException: Could not resolve module: org.jkiss.dbeaver.ui.statistics [276]
  Unresolved requirement: Require-Bundle: org.eclipse.core.runtime
    -> Bundle-SymbolicName: org.eclipse.core.runtime; bundle-version="3.27.0.v20230515-1719"; singleton:="true"
       org.eclipse.core.runtime [46]
         Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"
         Unresolved requirement: Require-Bundle: org.eclipse.core.contenttype; bundle-version="[3.8.0,4.0.0)"; visibility:="reexport"
           -> Bundle-SymbolicName: org.eclipse.core.contenttype; bundle-version="3.9.0.v20230412-0829"; singleton:="true"
              org.eclipse.core.contenttype [34]
                Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"
         Unresolved requirement: Require-Bundle: org.eclipse.core.jobs; bundle-version="[3.13.0,4.0.0)"; visibility:="reexport"
           -> Bundle-SymbolicName: org.eclipse.core.jobs; bundle-version="3.14.0.v20230317-0901"; singleton:="true"
              org.eclipse.core.jobs [42]
                Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"
  Unresolved requirement: Require-Bundle: org.eclipse.core.runtime
    -> Bundle-SymbolicName: org.eclipse.core.runtime; bundle-version="3.27.0.v20230515-1719"; singleton:="true"
…
```  
Similar issue with 23.1.1 (second error)

**Comment by jowilhnson** [comment]
Had the same issue on Manjaro Linux, but was able to resolve by changing the system's default java version.

`archlinux-java status` 
command to check what versions were installed and which was default (11)
 
then to change the default to a version 17 or newer that was already installed:
`archlinux-java set java-20-openjdk`

**Reply by Vricap** [reply]
This work! Thanks!

**Comment by bcag2** [comment]
Thanks @jowilhnson,
For Ubuntu 22, by running : 
`apt-cache search openjdk`  
to check available version, then run
`sudo apt-get install openjdk-19-jre`  (and openjdk-21-jre after update dbeaver from v25.1.0 to v25.1.1)
to install last JRE, dbeaver runs now correctly.  
NOTE : I installed only JRE (50Mo, 202Mo required on disk) instead of JDK (477Mo required on disk)
